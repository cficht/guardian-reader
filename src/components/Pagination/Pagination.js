import React from 'react';
import PropTypes from 'prop-types';
import './Pagination.scss';

export default function Pagination({ page, maxPage, handlePage }) {
  return (
    <section className="pagination-section">
      <button onClick={() => handlePage(-1)} disabled={page <= 1}>&larr;</button>
      <h3>{page} / {maxPage}</h3>
      <button onClick={() => handlePage(1)} disabled={page >= maxPage}>&rarr;</button>
    </section>
  );
}

Pagination.propTypes = {
  page: PropTypes.number,
  maxPage: PropTypes.number,
  handlePage: PropTypes.func
};
