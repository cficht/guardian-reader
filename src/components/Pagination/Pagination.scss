$primary-color: #052962;
$primary-font-color: #FFFFFF;

.pagination-section {
  display: flex;
  justify-content: center;
  width: 50%;
  margin: 20px;
  
  background-color: whitesmoke;

  button {
    width: 20%;
    margin: 20px;
    padding: 5px;
    color: $primary-font-color;
    background-color: $primary-color;
  }
}