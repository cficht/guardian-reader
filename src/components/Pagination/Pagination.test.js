/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import Pagination from './Pagination';

test('renders Pagination.js correctly', () => {
  const tree = renderer
    .create(<Pagination />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
