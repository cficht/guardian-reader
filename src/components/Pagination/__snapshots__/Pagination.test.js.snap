// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Pagination.js correctly 1`] = `
<section
  className="pagination-section"
>
  <button
    disabled={false}
    onClick={[Function]}
  >
    ←
  </button>
  <h3>
     / 
  </h3>
  <button
    disabled={false}
    onClick={[Function]}
  >
    →
  </button>
</section>
`;
